[WIP] fix Rancher page (#314)
portworx/pxdocs: content/install-with-other/rancher/rancher-2.x/_index.md

---
title: Portworx on Kubernetes using Rancher 2.x
linkTitle: Rancher 2.x
keywords: portworx, PX-Developer, container, Rancher, storage
description: Install Portworx on Kubernetes using Rancher 2.x with a public catalog (Helm Chart)
weight: 6
series: px-rancher
noicon: true
hideSections: true
---

> ## Purpose

Reviewer: I think we can skip "Purpose" and just jump straight into the concept information that follows. 

```suggestion

```

This guide walks through installing _Portworx_ on Kubernetes using Rancher 2.x with a Helm chart available from the public catalog.

> ## Installing Portworx on Kubernetes using Rancher 2.x

Reviewer: Why don't we make this the actual topic (page) title? It's basically the same title with better detail.

Author: @nathan-px Good idea!

> 

Reviewer: I know the older prerequisites needed to go, but I don't think we should have _no_ prerequisites. I'd add the following two points: 

## Prerequisites

* You must have a Kubernetes cluster with Rancher installed on it
* Your cluster must meet the [requirements](/start-here-installation/) for installing a Portworx cluster

Author: @nathan-px OK.

{{<info>}}

> Currently, `RancherOS` distro is not supported for _Portworx_.

Reviewer: ```suggestion
Currently, the `RancherOS` distro is not supported for _Portworx_.
```

{{</info>}}

Perform the following steps to install _Portworx_:

1. Navigate to your cluster. From the top navigation bar, select `Apps`. You will be taken to a page similar to the one shown below:

    ![Rancher Launch](/img/px-rancher-launch.png)

> 2. Click the "Launch" button and to then search for the _Portworx_ catalog.

Reviewer: Instead of "click" use the word "select". See the MMS for the different UI terms: https://docs.microsoft.com/en-us/style-guide/procedures-instructions/describing-interactions-with-ui

I've also started bolding buttons and UI text. 

```suggestion
2. Select the **Launch** button and then search for the _Portworx_ catalog.
```

3. Select `View Details` to start the Helm chart form:

    ![Get K8S Version](/img/px-rancher-1.png)

    _The content of the answer file is located in the appendix called answer.yml._

    **Storage Parameters**

    Set the drives field to any block device that you will be using for _Portworx_ storage.

    **Network Parameters**

    Type the name of the interface dedicated to _Portworx_ traffic in the `Data Network Interface` field (e.g.: `eth1`). Next, enter the Kubernetes host interface in the `Management Network Interface` field.

    **KVdb Options**

    If you want to use an external KVdb, enter your endpoints separated by semicolons. Use the following as an example: <br> `etcd://myetc1.company.com:2379;etcd://myetc2.company.com.2379`.

> 4. Once you've filled out the form, it should look similar to the one below:

Reviewer: This should just be part of the previous step. We're explaining what the fields are and how to populate them, then we can show a filled-out page example. Just like how we're doing it with spec examples. 
```suggestion

```

Author: Right. I removed it.

title: Portworx on Rancher 2.x
linkTitle: Rancher 2.x
keywords: portworx, PX-Developer, container, Rancher, storage

> description: Instructions on installing Portworx using public catalog (Helm Chart) on Rancher 2.x

Reviewer 2: same as above

Reviewer 2: install portworx on kubernetes *using* rancher

weight: 6
series: px-rancher
noicon: true
hideSections: true
---

## Purpose

> This guide walks through installing _Portworx_ on Rancher 2.x using a Helm chart available from the public catalog.

Reviewer 2: install portworx on kubernetes *using* rancher

> ## Installing Portworx on Rancher 2.x

Reviewer 2: install portworx on kubernetes *using* rancher

{{<info>}}
Currently, `RancherOS` distro is not supported for _Portworx_.
{{</info>}}

Perform the following steps to install _Portworx_:

> 1. Navigate to your cluster. From the top navigation bar, select `Apps`. You will be taken to a page similar to the one shown below:

Reviewer: This appears not to be visible from within the cluster view. Instead, you must be in the global cluster overview. 

If you go into a cluster dashboard, the top nav changes and `Apps` is no longer listed.

Author: @nathan-px I no longer have access to this Rancher cluster: https://70.0.0.38. If you can access a Rancher cluster, would be so so kind as to suggest how to update these sentences?

Reviewer: You screenshot is correct, I would just be more explicit about being in the global cluster overview page:

```suggestion
1. Navigate to the global cluster overview page. From the top navigation bar, select `Apps`. You will be taken to a page similar to the one shown below:
```

Author: @nathan-px Gotcha. Committed your suggestion.

Reviewer: @andreipope Did this commit get reverted somehow? I still see the original text.

Author: @nathan-px Have no idea. It should be fixed now.

    ![Rancher Launch](/img/px-rancher-launch.png)

2. Click the "Launch" button and to then search for the _Portworx_ catalog.

> 3. Select `View Details` to start the Helm chart form:

Reviewer: What about 
* name
* template version
* target projects
* cluster
* upgrade strategy
* available roles

Some of these are obvious, and I assume others are more Rancher's domain? Do we have any recommendations? We might consider adding a link to the step about the Rancher fields: https://rancher.com/docs/rancher/v2.x/en/catalog/multi-cluster-apps/

Author: @nathan-px I might be wrong, but I remember Harsh wanted to keep only the storage and network parameters.

Reviewer: Yeah, I guess most of these have default values. It makes sense that we don't want to document these if they're Rancher's domain.  

As someone who is not familiar with Rancher, it isn't clear to me what parameters are Rancher and what are Portworx, so linking out to them would help me. Something like this adds clarity without going into too much unnecessary detail:
```
3. Select `View Details` to start the Helm chart form:

     Configure the standard parameters for your cluster, visit the [Rancher documentation](https://rancher.com/docs/rancher/v2.x/en/catalog/multi-cluster-apps/#launching-a-multi-cluster-app) for more information.

     Configure the network and storage parameters...
```

@harsh-px thoughts?

Reviewer: @andreipope Looking at this again, I think it's helpful to address what params are Rancher specific and what params they need to configure as part of Portworx.

Author: @nathan-px I suspect these parameters are Portworx specific, and they seem pretty self-explanatory. Just take a look at the screenshot and let me know your thoughts.

Reviewer: It isn't immediately clear to me that we're asking users to perform steps. I think this can be handled with better formatting. 

Even though the UI is pretty easy to follow, I don't feel like the instructions are providing any answers to questions I may have. The picture also doesn't show the full page of what I'll see (all the Rancher-specific stuff). So I still think a prompt pointing users to relevant Rancher doc is helpful and brief. 

I think Harsh's intent is to streamline the process to make the deployment as easy as possible, and while the defaults do that quite well, the documentation doesn't improve by omitting information. Instead, I end up feeling uncertain as I go through the process and see more fields than I expected. 

I'm taking a stab at formatting here: 

```suggestion
3. Select `View Details` to start the Helm chart form:

     * Configure the standard parameters for your cluster, visit the [Rancher documentation](https://rancher.com/docs/rancher/v2.x/en/catalog/multi-cluster-apps/#launching-a-multi-cluster-app) for more information.
     * Configure the following Portworx parameters:
        * **Storage Parameters**: Set the drives field to any block device that you will be using for Portworx storage.
        * **Network Parameters**: Enter the name of the interface dedicated to Portworx traffic in the **Data Network Interface** field (e.g.: `eth1`). Next, enter the Kubernetes host interface in the **Management Network Interface** field.
        * **KVDB Options**: If you want to use an external KVDB, enter your endpoints separated by semicolons. Use the following as an example: 
etcd://myetc1.company.com:2379;etcd://myetc2.company.com.2379.
```

We might want to do a quick call to discuss this and try a few different things.

Author: @nathan-px I pushed the changes you suggested.

    ![Get K8S Version](/img/px-rancher-1.png)

    _The content of the answer file is located in the appendix called answer.yml._

>     **Storage Parameters**

Reviewer: What about KVDB and security? If we have relevant sections for configuring those in our docs, we should link to them.

Author: @nathan-px This is part of the initial requirements -> "Don't talk about KVDB in the docs... the helm chart will cover this". If you think it makes sense to add more info, we should check with Harsh first.

Reviewer: I see it now in the ticket. So this field will be populated with default values in the production version? @harsh-px

Author: @trierra The link points to a section called "Installing Portworx on AWS". Please, can you be more specific about what needs to be included in the docs?

>     Set the drives field to any block device that you will be using for _Portworx_ storage. A recommended practice is to add a separate SSD block device as a Journal drive.

Author: > > A recommended practice is to add a separate SSD block device as a Journal drive.
> 
> can remove this

Done.

    **Network Parameters**

> 

Author: @trierra @nathan-px @harsh-px We should clarify this. The initial requirement was to remove existing references to kvdb.

> 

Reviewer: I think over-documenting a little bit, in this case, is less likely to confuse readers than leaving the defaults and hoping they're adequate. Maybe include a description of what the default is, and what happens if you leave it as well.

> 

Author: @trierra @nathan-px It's updated.

title: Portworx on Rancher 2.x
linkTitle: Rancher 2.x
keywords: portworx, PX-Developer, container, Rancher, storage
description: Instructions on installing Portworx using public catalog (Helm Chart) on Rancher 2.x
weight: 6
series: px-rancher
noicon: true
---

This guide walks through installing _Portworx_ on Rancher 2.x using a Helm chart available from the public catalog.

> ## Pre-requisites

Reviewer 3: Remove  Pre-requisites altogether. We have a Prerequisites thats at the top of the left sidebar. Users can just go there.

Author: > Remove Pre-requisites altogether. We have a Prerequisites thats at the top of the left sidebar. Users can just go there.

done

- Make sure your environment meets the [minimum requirements](/start-here-installation/#installation-prerequisites).
- Rancher 2.x cluster that runs Kubernetes version 1.10 or higher. You can install Rancher 2.x by following [these instructions](https://rancher.com/docs/rancher/v2.x/en/installation/).

## Installing Portworx on Rancher 2.x

{{<info>}}
Currently, `RancherOS` distro is not supported for _Portworx_.
{{</info>}}

> _Portworx_ provides a Helm chart for Rancher 2.x that is available from the public catalog. The Helm chart deploys a trial version of the _Portworx_ Enterprise Edition that you can use for 30 days. After the trial version expires, you must [purchase a Portworx license](/reference/knowledge-base/px-licensing/) to continue to use your _Portworx_ cluster.

Reviewer 3: Remove this paragraph.

Author: done

> In addition, the Helm chart will also deploy [Stork](https://github.com/libopenstorage/stork) to your Kubernetes cluster. Stork is the _Portworx_ storage scheduler. It allows you to co-locate pods with their data, and create and restore snapshots of _Portworx_ volumes.

Reviewer 3: remove this Stork paragraph.

Author: done

To install the _Portworx_ Helm chart, navigate to your cluster. Then, select the `System` namespace.  Search for the _Portworx_ catalog in the load application section and select `View Details` to start the Helm chart form.  The contents of the answer file are located in the appendix called answer.yml.

![Get K8S Version](/img/px-rancher-1.png)

### Storage Parameters

> In your environment, set the drives field to any block device that you will be using for _Portworx_ storage. A recommended practice is to add a separate SSD block device as a Journal drive. If you have one available, enter auto in the Journal device section.

Reviewer 4: Journal device section will be set to `auto` by default with no option to change. Last sentence is no longer needed

Reviewer 4: @harsh-px do we need the `Configure Metadata Device` section  for KVDB here?

Author: removed "If you have one available, enter auto in the Journal device section."

### Network Parameters

In your environment, you will put the interface dedicated to _Portworx_ traffic in the `Data Network Interface` field. Next, enter the Kubernetes host interface in the `Management Network Interface`.

> ### Advanced Parameters

Reviewer 3: Remove `Advanced Parameters` altogether. All defaults just work.

Author: done

- Set the `Install Stork` and `Lighthouse` fields to `true`.
- Make sure you give the cluster a name that is relevant to your environment.

> - Set the following version information:

Reviewer 3: Remove this `Set the following version information:`. The catalog already has the defaults here.

Author: done
